modules/partition-replicator/src/main/java/org/apache/ignite/internal/partition/replicator/PartitionReplicaLifecycleManager.java

```diff
@@ -1257,6 +1219,58 @@ private CompletableFuture<Boolean> isLocalNodeIsPrimary(ReplicationGroupId repli
         }
     }
 
+    private void executeIfLocalNodeIsPrimaryForGroup(
+            ReplicationGroupId groupId,
+            Consumer<ReplicaMeta> toExecute
+    ) {
+        CompletableFuture<ReplicaMeta> primaryReplicaFuture = getPrimaryReplica(groupId);
+
+        isLocalNodeIsPrimary(primaryReplicaFuture).thenAccept(isPrimary -> {
```

Comment:
You don't have any exception handling here, you will simply swallow them if any of the futures fail for some reason